imap/jmap_api.h

```diff
@@ -171,11 +171,19 @@ typedef struct {
     const char *from_accountid;  // input to the handler
     const char *blobid;          // input to the handler
     const char *accept_mime;     // input to the handler
+    unsigned decode : 1;         // input to the handler
     struct buf blob;             // output from the handler
-    const char *content_type;    // output from the handler
+    char *content_type;          // output from the handler
+    char *encoding;              // output from the handler
     const char *errstr;          // output from the handler
 } jmap_getblob_context_t;
 
+#define GETBLOB_CTX_INITIALIZER(from_accountid, blobid, accept_mime, decode) \
```

Comment:
This deviates from the currently used idioms in the codebase: None of the current INITIALIZER macros takes arguments. Instead they initialize a struct with default values that make the data structure ready for use without further initialization. (And the way the macro currently is defined is not safe with regards to potential duplicate evaluation).

For parameterized initializers there's a couple of foo_init functions, but unfortunately they do not all follow the same pattern. Some are void functions, others return an initialized struct. I tend to use void functions.

For the particular case, though, I would just not use an INITIALIZER at all. Is the blob context initialized at more than one code section?

Reply:
Fixed in latest commit